test/unit/io/sequence_file/sequence_file_input_test.cpp

```diff
@@ -378,3 +405,91 @@ TEST_F(sequence_file_input_f, column_reading_decomposed_temporary)
         EXPECT_TRUE(empty(quals[i]));
     }
 }
+
+// ----------------------------------------------------------------------------
+// decompression
+// ----------------------------------------------------------------------------
+
+template <typename fixture_t, typename input_file_t>
+void decompression_impl(fixture_t & fix, input_file_t & fin)
+{
+    size_t counter = 0;
+    for (auto & rec : fin)
+    {
+        EXPECT_TRUE((std::ranges::equal(get<field::SEQ>(rec), fix.seq_comp[counter])));
+        EXPECT_TRUE((std::ranges::equal(get<field::ID>(rec),  fix.id_comp[counter])));
+        EXPECT_TRUE(empty(get<field::QUAL>(rec)));
+
+        counter++;
+    }
+
+    EXPECT_EQ(counter, 3u);
+}
+
+#ifdef SEQAN3_HAS_ZLIB
+std::string input_gz
+{
+    '\x1F','\x8B','\x08','\x00','\x33','\xBF','\x13','\x5C','\x00','\x03','\xB3','\x53','\x08','\x71','\x0D','\x0E',
+    '\x51','\x30','\xE4','\x72','\x74','\x76','\x0F','\xE1','\xB2','\x0B','\x49','\x2D','\x2E','\x31','\xE2','\x72',
+    '\x74','\x77','\x77','\x0E','\x71','\xF7','\xE3','\xB2','\x53','\x00','\xF1','\x8D','\xB9','\xDC','\xDD','\x1D',
+    '\xDD','\x43','\x1C','\x43','\x1C','\x1D','\x43','\x50','\x21','\x17','\x00','\xEF','\x24','\xC2','\xE9','\x3E',
+    '\x00','\x00','\x00',
```

Comment:
You shouldn't write your master password in plain text. 😁

Reply:
:male_detective: